src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -2605,6 +2605,237 @@ static S1 MayWrap(ref Span<int> arg)
+                    {
+                        Span<int> stackSpan = stackalloc int[] { 13 };
+                        Span<int> heapSpan = default;
+                        S1 local = default;
+                        _ = local[stackSpan]; // 1
+                        _ = local[heapSpan];
+                    }
+                }
+
+                ref struct S2
+                {
+                    public readonly int this[Span<int> span] => 42;
+
+                    static void Test()
+                    {
+                        Span<int> stackSpan = stackalloc int[] { 13 };
+                        Span<int> heapSpan = default;
+                        S2 local = default;
+                        _ = local[stackSpan];
+                        _ = local[heapSpan];
+                    }
+                }
+
+                ref struct S3
+                {
+                    public int this[Span<int> span]
+                    {
+                        get => 42;
+                        set { }
+                    }
+
+                    static void Test()
+                    {
+                        Span<int> stackSpan = stackalloc int[] { 13 };
+                        Span<int> heapSpan = default;
+                        S3 local = default;
+                        _ = local[stackSpan]; // 2
+                        _ = local[heapSpan];
+                        local[stackSpan] = 42; // 3
+                        local[heapSpan] = 42;
+                    }
+                }
+
+                ref struct S4
+                {
+                    public int this[Span<int> span]
+                    {
+                        readonly get => 42;
+                        set { }
+                    }
+
+                    static void Test()
+                    {
+                        Span<int> stackSpan = stackalloc int[] { 13 };
+                        Span<int> heapSpan = default;
+                        S4 local = default;
+                        _ = local[stackSpan];
+                        _ = local[heapSpan];
+                        local[stackSpan] = 42; // 4
+                        local[heapSpan] = 42;
+                    }
+                }
+
+                ref struct S5
+                {
+                    public int this[Span<int> span]
+                    {
+                        get => 42;
+                        readonly set { }
+                    }
+
+                    static void Test()
+                    {
+                        Span<int> stackSpan = stackalloc int[] { 13 };
+                        Span<int> heapSpan = default;
+                        S5 local = default;
+                        _ = local[stackSpan]; // 5
+                        _ = local[heapSpan];
+                        local[stackSpan] = 42;
+                        local[heapSpan] = 42;
+
+                        local[stackSpan]++; // 6
+                        local[heapSpan]++;
```

Comment:
Should we test `local[stackSpan]++; local[heapSpan]++;` for `S3` and `S4` as well?